lua/neotest-vstest/dotnet_utils.lua

```diff
@@ -344,4 +331,57 @@ function dotnet_utils.build_project(project)
   return dotnet_utils.build_path(project.proj_file)
 end
 
+local sysname = vim.uv.os_uname().sysname:lower()
+local iswin = not not (sysname:find("windows") or sysname:find("mingw"))
+
+--- Attempts to extract the project path from a line in a solution file
+---@param line string
+---@param target string
+---@return string? path The path to the project file
+local function sln_match(line, target)
+  local ext = vim.fn.fnamemodify(target, ":e")
+
+  if ext == "sln" then
+    local id, name, path = line:match('Project%("{(.-)}"%).*= "(.-)", "(.-)", "{.-}"')
+    if id and name and path and path:match("%.[cf]sproj$") then
+      return path
+    end
+  elseif ext == "slnx" then
+    local path = line:match('<Project Path="([^"]+)"')
+    if path and path:match("%.[cf]sproj$") then
+      return path
+    end
+  elseif ext == "slnf" then
+    return line:match('"(.*%.[cf]sproj)"')
+  else
+    error(string.format("Unknown extension `%s` for solution: `%s`", ext, target))
+  end
+end
+
+---@param target string Path to solution or solution filter file
+---@return string[] Table of projects in given solution
+function dotnet_utils.projects(target)
+  local file = io.open(target, "r")
```

Comment:
I think we should use the `nio` file functions here to be consistent with the rest of the code base
```suggestion
  local file = nio.file.open(target, "r")
```